python/social_bot/tasks.py

```diff
@@ -1137,3 +1142,249 @@ def task_specific_observation(self, agent):
              palm_pos]).flatten()
         return np.concatenate(
             (obs, finger_contacts, agent_pose, joints_states), axis=0)
+
+
+@gin.configurable
+class Stack(Task):
+    """
+    A task to stack several wood cubes together.
+    """
+
+    compatible_agents = ['youbot_noplugin']
+
+    def __init__(self,
+                 env,
+                 max_steps,
+                 max_distance=0.5,
+                 min_distance=0.3,
+                 objects_num=3,
+                 objects_random_range=0.25,
+                 object_half_size=0.025,
+                 success_distance_thresh=0.03,
+                 reward_shaping=True,
+                 reward_weight=1.0):
+        """
+        Args:
+            env (gym.Env): an instance of Environment
+            max_steps (int): episode will end if not complet the task in so many steps, recommend to be 150
+                for agent youbot_noplugin and object 5cm cube
+            max_distance (float): the max distance from the random object's location to the agent
+            min_distance (float): the min distance from the random object's location to the agent
+            objects_num (int): the number of objects to stacking
+            objects_random_range (float): the range of objects distributed
```

Comment:
There is already max and min_distance, why is this random_range needed?

Reply:
max and min_distance is the distance from central position of the distribution of objects to the agent. objects_random_range is the range of objects from the central position.

Reply:
Ah, objects are distributed around a center according to objects_random_range, and those positions that fall outside of agent's max_distance will be eliminated?

In that case, if agent gets too far away from object distribution's center, then no object will be placed in the env?

Reply:
objects outside of agent's max_distance will not be eliminated... 

max_distance and min distance is only the range for the center position. Maybe rename to max_center_distance or something would be better?

Reply:
Oh, I see.  Please add this detail to the comment (how objects are initialized).  Name can be something like max_distance_from_center and object_center_random_range?